oemoflex/model/variations.py

```diff
@@ -1,30 +1,185 @@
+    these intervals, different sampling methods can be invoked.
+    """
+    AUX_COLUMNS = ["carrier", "region", "tech", "type"]
+    VAR_VALUE = "var_value"
 
-            _dp.data[resource].loc[:, var_name] = var_value
+    def __init__(self, lb, ub, eps=1e-6):
+        self.lb = lb
+        self.ub = ub
+        self.eps = eps
+        self.sanity_check()
 
-        return _dp
+    def sanity_check(self):
+        # assert index same
+        assert index_same(self.lb, self.ub), "Indexes of lb and ub are not the same."
+
+        # Assert that the data is different
+        assert not data_same(
+            self.lb, self.ub
+        ), "There is no difference between lb and ub."
+
+        # Assert that the auxiliary columns are the same
+        assert data_same(
+            self.lb.loc[:, self.AUX_COLUMNS], self.ub.loc[:, self.AUX_COLUMNS]
+        )
+
+        # find the parameters that are different (comparing NaN)
+        diff = self.get_diff()
+
+        # assert lb <= ub
+        assert smaller_equal(
+            self.lb.loc[diff, self.VAR_VALUE], self.ub.loc[diff, self.VAR_VALUE]
+        )
+
+        # assert that the difference is larger than eps
+
+        assert diff_larger_eps(
+            self.lb.loc[diff, self.VAR_VALUE],
+            self.ub.loc[diff, self.VAR_VALUE],
+            self.eps,
+        ).all(), f"The difference between lb and ub is lower than the defined minimum of {self.eps}"
+
+    def get_diff(self):
+        return get_diff(self.lb.loc[:, self.VAR_VALUE], self.ub.loc[:, self.VAR_VALUE])
+
+    def get_param(self):
+        # get the parameters that are varied
+        param = self.lb.index[self.get_diff()]
+        return param
+
+    def get_linear_slide(self, n):
+
+        params = self.get_param()
+
+        # for each param, create n samples
+        samples = {}
+
+        for i in range(n):
+            sample = self.lb.copy()
+            slider = i / (n - 1)
+            sample.loc[params, "var_value"] = (1 - slider) * self.lb.loc[
+                params, "var_value"
+            ] + slider * self.ub.loc[params, "var_value"]
+            sample_name = "sample_" + str(i)
+            samples[sample_name] = sample
+
+        return samples
+
+    def get_salib_samples(self, salib_method, **kwargs):
+
+        params = self.get_param()
+
+        problem = {
+            "num_vars": len(params),
+            "names": list(params),
+            "bounds": list(
+                zip(self.lb.loc[params, "var_value"], self.ub.loc[params, "var_value"])
+            ),
+        }
+
+        samples = salib_method(problem, **kwargs)
+
+        full_samples = {}
+        for i, sample in enumerate(samples):
+            full_sample = self.lb.copy()
+            full_sample.loc[params, "var_value"] = sample
+            sample_name = "sample_" + str(i)
+            full_samples[sample_name] = full_sample
+
+        return full_samples
+
+
+class EDPSensitivity(Sensitivity):
```

Comment:
This could be done more general. The above Sensitivity class does checking and sampling for a single dataframe. This class is supposed to do the same for DataFramePackages.

The implementation takes some time to grasp - could be more transparent?